### PR TITLE
add()/update() on Property controller/service (#23)

### LINKED ISSUE #23: To implement add Property (service/controller)
- to add REST/PUT method to the PropertyController
- add() method in the PropertyService
- unit tests

### PR DESCRIPTION
**Pull request by egch**
Tested with postman ✔️ 
- added id to the Property DTO
- add(): service/controller
- update() :service/controller
- unit tests: service/controller

### CODE REVIEW

File: your-rents-core/src/main/java/com/yourrents/core/dto/Property.java [hunk 1]
```diff
@@ -20,13 +20,21 @@
...
+import lombok.Setter;
 import lombok.ToString;
 
 @Builder
 @Getter
 @ToString
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(of = {"name"})
```
Comment (benfante) [inline your-rents-core/src/main/java/com/yourrents/core/dto/Property.java]: mmmm...no, name it's a brief description, not an id. you should use external_id.

File: your-rents-services/src/main/java/com/yourrents/services/controller/PropertyController.java [hunk 2]
```diff
@@ -43,4 +45,13 @@
         return ResponseEntity.ok(propertyService.list(pageable));
     }
 
+    @RequestMapping(
+            method = {RequestMethod.PUT},
```
Comment (benfante) [inline your-rents-services/src/main/java/com/yourrents/services/controller/PropertyController.java]: PUT? It'should be POST for adding a new data. PUT is for modify.

File: your-rents-services/src/main/resources/logback-spring.xml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  #L%
+  -->
+
+<configuration>
+
+    <include resource="org/springframework/boot/logging/logback/default.xml"/>
```
Comment (benfante) [inline your-rents-services/src/main/resources/logback-spring.xml]: There is a reason why you are importing spring boot default, then overiding it and not using defaults at all?

**Review by benfante — CHANGES_REQUESTED** [review]

File: your-rents-core/src/main/java/com/yourrents/core/dto/Property.java [hunk 4]
```diff
@@ -20,14 +20,23 @@
...
 
 @Builder
 @Getter
 @ToString
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(of = {"id"})
 public class Property {
+    private Integer id;
```
Comment (benfante) [inline your-rents-core/src/main/java/com/yourrents/core/dto/Property.java]: Do you like to expose the id in the external layer? I would like more to use the external_id.

**Review by benfante — CHANGES_REQUESTED** [review]

File: your-rents-core/src/main/java/com/yourrents/core/YRNotFoundException.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+public class YRNotFoundException extends RuntimeException {
```
Comment (benfante) [inline your-rents-core/src/main/java/com/yourrents/core/YRNotFoundException.java]: Qual è il vantaggio di chiamarla YRNotFoundException? Non basta NotFoundException?

**Review by benfante — CHANGES_REQUESTED** [review]

File: your-rents-core/src/main/java/com/yourrents/core/YRNotFoundException.java (hunk 5 above)
Comment (egch) [reply your-rents-core/src/main/java/com/yourrents/core/YRNotFoundException.java]: Sure, you’re right, I’ll change it

**Review by benfante — APPROVED** [review]

**Pull request merged**